agent/engine/docker_container_engine.go

```diff
@@ -743,7 +747,26 @@ func metadataFromContainer(dockerContainer *docker.Container) DockerContainerMet
 	if dockerContainer.State.OOMKilled {
 		metadata.Error = OutOfMemoryError{}
 	}
+	if dockerContainer.State.Health.Status == "" {
+		return metadata
+	}
+
+	// Record the health check information if exists
+	health := api.HealthStatus{}
+	logLength := len(dockerContainer.State.Health.Log)
+	if logLength != 0 {
+		// Only save the last log from the health check
+		health.Output = dockerContainer.State.Health.Log[logLength-1].Output
```

Comment:
> Is this only the last line from the health check log? will this have enough context?
Yes, since the log is the health check command output, and if the command failed the output should be the same. So I think the last line is enough.

>is this the only way to get visibility into why the health check command may have failed?

You can also get the failure reason from exitcode. the output is the health check command output, which is more understandable than the exitcode.

Reply:
Why would you only have the last line?  Also: I don't see any limit here to the length of the line.  We should probably limit this in terms of bytes and maybe select the last N bytes instead of the last line.